Guys, I can't see a workaround, it's working in the shell but failing in the test script:

```
class CustomNA:
	def __bool__(self):
		raise TypeError("CustomNA cannot be interpreted as a boolean value.")

	def __eq__(self, other):
		return self
CustomNA() in np.array(["a"], dtype=object)
```
>  DeprecationWarning: elementwise comparison failed; this will raise an error in the future.

In shell:
```
>>> CustomNA() not in np.array(["a"], dtype=object)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in __bool__
TypeError: CustomNA cannot be interpreted as a boolean value.
```

Can you please help me. Apologies if it's a very basic doubt.

--- a/numpy/core/src/multiarray/sequence.c
+++ b/numpy/core/src/multiarray/sequence.c
@@ -38,9 +38,9 @@ array_contains(PyArrayObject *self, PyObject *el)
     if (res == NULL) {
         return -1;
     }
-    Py_DECREF(res);
 
     any = PyArray_Any((PyArrayObject *)res, NPY_MAXDIMS, NULL);
+    Py_DECREF(res);
     if (any == NULL) {
         return -1;
     }

